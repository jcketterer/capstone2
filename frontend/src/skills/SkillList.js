import React, { useContext, useState, useEffect, useCallback } from 'react';
import { Redirect } from 'react-router-dom';
import UserContext from '../UserContext';
import SkillsCard from './SkillsCard';
import AdvocateAPI from '../api';

const SkillList = () => {
  const user = useContext(UserContext);
  const [skills, setSkills] = useState([]);
  const [filter, setFilter] = useState('');

  useEffect(() => {
    let isRendered = true;

    async function getSkills(filterString = '') {
      try {
        if (isRendered && user.username) {
          let skillRes = await AdvocateAPI.getSkillByName(filterString);
          setSkills(skillRes);
        }
      } catch (err) {
        console.log(err);
      }
    }
    getSkills(filter);

    return () => {
      isRendered = false;
    };
  }, [filter, user.username]);

  if (!user.username) {
    return <Redirect to="/" />;
  }

  return (
    <div className="SkillList">
      <div className="container">
        <h3 className="display-3 mb-3">Skills:</h3>
        <div className="d-inline-flex flex-column">
          {skills.length > 0 ? (
            skills.map(skill => (
              <SkillsCard skill={skill} key={skill.skillName} name={skill.name} />
            ))
          ) : (
            <p>No skills found.</p>
          )}
        </div>
      </div>
    </div>
  );
};

export default SkillList;
