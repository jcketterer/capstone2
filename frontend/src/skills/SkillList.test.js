import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import SkillList from './SkillList';
import UserContext from '../UserContext';

it('should render without breaking', () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <SkillList />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it('matches the snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <SkillList />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
