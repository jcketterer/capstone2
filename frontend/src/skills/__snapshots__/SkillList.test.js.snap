// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="SkillList"
  >
    <div
      class="container"
    >
      <h3
        class="display-3 mb-3"
      >
        Skills:
      </h3>
      <div
        class="d-inline-flex flex-column"
      >
        <p>
          No skills found.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
